src/components/tools/tool-tile.tsx

```diff
@@ -146,24 +163,34 @@ export class ToolTileComponent extends BaseComponent<IProps> {
   }
 
   public render() {
-    const { model, widthPct } = this.props;
+    const { model, readOnly, widthPct } = this.props;
+    const { hoverTile } = this.state;
     const { appConfig, ui } = this.stores;
-    const selectedClass = ui.isSelectedTile(model) ? " selected" : "";
     const ToolComponent = kToolComponentMap[model.content.type];
     const isPlaceholderTile = ToolComponent === PlaceholderToolComponent;
-    const placeholderClass = isPlaceholderTile ? " placeholder" : "";
-    const dragTileButton = !isPlaceholderTile && !appConfig.disableTileDrags &&
-                            <div ref={elt => this.dragElement = elt}><DragTileButton /></div>;
+    const isTileSelected = ui.isSelectedTile(model);
+    const classes = classNames("tool-tile", {
+                      placeholder: isPlaceholderTile,
+                      readonly: readOnly,
+                      hovered: this.state.hoverTile,
+                      selected: isTileSelected });
```

Comment:
Nice tool for constructing these long CSS class names.  Probably useful in some other projects.

Reply:
I've been tempted to use it in a number of other places, but there's a slippery slope issue at work. If there's only one conditional class (e.g. `${condition ? class : ""}`) it doesn't seem worth it to introduce the library. And if there's already one conditional class, adding one more seems simpler than introducing the library.... In this case, the library is already in the project, and moving from two conditional classes to four clearly meets the criteria for use of the library. But I'm sure there are other places where it would be useful as well.